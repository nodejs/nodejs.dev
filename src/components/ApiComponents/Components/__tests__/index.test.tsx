import React from 'react';
import { render } from '@testing-library/react';
import { ApiLink, H5, H4, H3, JsonLink } from '..';

describe('ApiComponents Components', (): void => {
  it('renders ApiLink component correctly', () => {
    const { container } = render(
      <ApiLink href="/test/datatag-tagc--">
        <br />
      </ApiLink>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders H5 component correctly', () => {
    const { container } = render(<H5 id="/test/tag-tagm--" />);
    expect(container).toMatchSnapshot();
  });

  it('renders H4 component correctly', () => {
    const id = '<DataTag tag="M" />';
    const { container } = render(<H4 id={id} />);
    expect(container).toMatchSnapshot();
  });

  it('renders H3 component correctly', () => {
    const id = '<Tag tag="C" />';
    const { container } = render(<H3 id={id} />);
    expect(container).toMatchSnapshot();
  });

  it('renders JsonLink component correctly', () => {
    const { container } = render(<JsonLink fileName="cli" version="v18" />);
    expect(container).toMatchSnapshot();
  });
});
