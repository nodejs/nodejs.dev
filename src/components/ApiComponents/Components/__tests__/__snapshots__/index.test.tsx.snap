// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiComponents Components renders ApiLink component correctly 1`] = `
<div>
  <a>
    <br />
  </a>
</div>
`;

exports[`ApiComponents Components renders H3 component correctly 1`] = `
<div>
  <h3
    id="<Tag tag="C" />"
  />
</div>
`;

exports[`ApiComponents Components renders H4 component correctly 1`] = `
<div>
  <h4
    id="<DataTag tag="M" />"
  />
</div>
`;

exports[`ApiComponents Components renders H5 component correctly 1`] = `
<div>
  <h5
    id="/test/"
  />
</div>
`;

exports[`ApiComponents Components renders JsonLink component correctly 1`] = `
<div>
  <div
    class="json"
  >
    <a
      href="https://nodejs.org/docs/latest-v18.x/api/cli.json"
    >
      <span>
        View as JSON
      </span>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-robot "
        data-icon="robot"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M320 0c17.7 0 32 14.3 32 32V96H472c39.8 0 72 32.2 72 72V440c0 39.8-32.2 72-72 72H168c-39.8 0-72-32.2-72-72V168c0-39.8 32.2-72 72-72H288V32c0-17.7 14.3-32 32-32zM208 384c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16H208zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16H304zm96 0c-8.8 0-16 7.2-16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s-7.2-16-16-16H400zM264 256a40 40 0 1 0 -80 0 40 40 0 1 0 80 0zm152 40a40 40 0 1 0 0-80 40 40 0 1 0 0 80zM48 224H64V416H48c-26.5 0-48-21.5-48-48V272c0-26.5 21.5-48 48-48zm544 0c26.5 0 48 21.5 48 48v96c0 26.5-21.5 48-48 48H576V224h16z"
          fill="currentColor"
        />
      </svg>
    </a>
  </div>
</div>
`;
